src/main/java/com/gmail/goosius/siegewar/playeractions/PlunderTown.java

```diff
@@ -231,4 +281,23 @@ private static void transferPlunderToNation(Siege siege, Nation nation, double t
 			}
 		}
 	}
+	
+	private static Map<Resident, Integer> gatherResidentsShareMap(Siege siege) {
+		Resident resident;
+		Map<Resident, Integer> residentSharesMap = new HashMap<>();
+		for(Map.Entry<String, Integer> uuidShareMapEntry: siege.getAttackerSiegeContributors().entrySet()) {
+			resident = TownyUniverse.getInstance().getResident(UUID.fromString(uuidShareMapEntry.getKey()));
+			if(resident != null)
+				residentSharesMap.put(resident, uuidShareMapEntry.getValue());
+		}
+		return residentSharesMap;
+	}
+	
+	private static double getWinnerRatio(double totalPlunderAmount) {
```

Comment:
:construction: 
- I like this idea to modularize the logic !
- First, one small change - please rename method to "getTotalPlunderForNationBank()"
- The key concept for the method, is to calculate the amount which will go to the nation bank.
- Then the logic needs to be revised
- This following should work (note the variable name improvement over current code, which may assist)
```
		String[] distributionRatios = distributionRatio.split(":");  //This includes both soldier and bank ratios
                                                                                                         //Today, nation **and** allied soldiers contribute
		int bankRatio = Integer.parseInt(distributionRatios [0]);
		int soldierRatio = Integer.parseInt(distributionRatios [1]);
		int totalRatio = bankRatio + soldierRatio;
		return totalPlunderAmount / totalRatio * bankRatio;
```